alpha-build-machinery/make/targets/openshift/crd-schema-gen.mk

```diff
@@ -0,0 +1,25 @@
+CRD_SCHEMA_GEN_VERSION ?=v0.2.1
+CRD_SCHEMA_GEN_APIS ?=./vendor/github.com/openshift/api/operator/v1
+
+# $1 - schema gen version
+# $2 - manifests directory
+# $3 - package APIs
+crd-schema-gen:
+	mkdir -p $(CRD_SCHEMA_GEN_TEMP)/bin
+	git clone -b $(CRD_SCHEMA_GEN_VERSION) --single-branch --depth 1 https://github.com/kubernetes-sigs/controller-tools.git $(CRD_SCHEMA_GEN_TEMP)/gen
+	cd $(CRD_SCHEMA_GEN_TEMP)/gen; GO111MODULE=on go build ./cmd/controller-gen; mv controller-gen ../bin
+	curl -f -L -o $(CRD_SCHEMA_GEN_TEMP)/bin/yq https://github.com/mikefarah/yq/releases/download/2.4.0/yq_$(shell uname -s | tr A-Z a-z)_amd64 && chmod +x $(CRD_SCHEMA_GEN_TEMP)/bin/yq
+.PHONY: crd-schema-gen
+
+update-codegen-crds: CRD_SCHEMA_GEN_TEMP :=$(shell mktemp -d)
```

Comment:
I'd make it a top level var so you can force it to use the same dir and would detect not to install if present

Reply:
I guess let's install it by default into `_output/tools/` so we cache it

Reply:
I originally had it as a top level var and the tool would fail for me. We also tried top level before this switch and ended up needing to make it a dep to the target like this